### ISSUE TITLE
"fmpq not defined" error

### ISSUE THREAD

**Issue by johannes-hoffmann** [oscar-system/Singular.jl#148]
In a rather complicated use case involving both Nemo and Singular, I encountered `ERROR: UndefVarError: fmpq not defined` pointing to line 165 in the following function:

https://github.com/oscar-system/Singular.jl/blob/af812af346209fee3c54179c3763e9f390c92f8d/src/libsingular/flint/fmpq.jl#L164-L168

Judging from other functions in the neighborhood, there seem to be some `Nemo.` missing. I can try to make a PR, but there could be some other instances like this you might want to fix in one go.

**Comment by wbhart** [comment]
A PR would be great. If you could search for all fmpz and fmpq's in that file, it would be a great help.

**Comment by wbhart** [comment]
In theory I already fixed all these. But I must have missed this particular file for some reasons. It's a wonder it passes the tests.

If you could add a test to trigger this case, that would be even more helpful. The tests are in the file [1] if I am not mistaken.

[1] https://github.com/oscar-system/Singular.jl/blob/master/test/libsingular/nemo-test.jl

**Comment by johannes-hoffmann** [comment]
Something like
```
using Singular
using Nemo
SR, (a,b) = Singular.PolynomialRing(Nemo.QQ, ["a", "b"])
std(Ideal(SR, a*b-1, a*a))
```
reproduces the error.

Basically all `fmpq` should be `Nemo.fmpq`? And the same for `fmpz` in the file next door?

**Comment by wbhart** [comment]
Yeah that looks harmless enough. If you could add that to the relevant test, that would be fantastic.

Yes, all fmpq should be Nemo.fmpq and fmpz should be Nemo.fmpz. Since you are going to fix a few of these, you may as well fix them all throughout the file. Most of them should be correct, but I see a few others which are not, especially when used as type assertions (::fmpq or ::fmpz).

**Comment by johannes-hoffmann** [comment]
I will open a PR soon. What about the call to `fmpq` in line 167?

**Comment by wbhart** [comment]
Fixed by #149
